staging/src/k8s.io/apiserver/pkg/server/options/authentication.go

```diff
@@ -115,6 +115,7 @@ func (s *RequestHeaderAuthenticationOptions) ToAuthenticationRequestHeaderConfig
 	if err != nil {
 		return nil, err
 	}
+	go caBundleProvider.Run(1, context.TODO().Done())
```

Comment:
leaking go routines looks wrong.  Some effort was taken to be able to plumb these back into post-starthooks to avoid leaking.  Why not track where that fell down and/or make the poststarthook something set up directly in this method and returned.

Reply:
ok, I'll see if I can track all the instances created and start them in the remote places.